### DISCUSSION TITLE
OKD 4.14.0 failed to connect to the openshift console (1 master / 3 workers)

### DISCUSSION THREAD

**Discussion by MainMan1998** [okd-project/okd#1823] [Q&A] [answered]
Hello, I'm trying to install okd 4.14.0-0.okd-2023-11-14-101924 on bare metal server using Fedora Core & Red hat linux.
I try to install it with 1 master and 3 computes nodes.
I have access to the cluster, I can do some oc commands, the master / worker are ready.
When I try to connect to the openshift console, I receive the following error: ERR_CONNECTION_CLOSED

Master: 32GB ram / 8 cpu
Worker: 16gb ram / 4 cpu
Okd4 version: 4.14.0-0.okd-2023-11-14-101924
Fedora Core Version: 38.20231002.3.1

The cluster is up on the redhat website.
After some hour of research, I remark that the authentication is running but the service seems to be down/not responding.

I attached some logs:

- oc adm must-gather:

When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
ClusterID: 3c298b2a-db03-4dc7-a7a2-1b6a3a98d8ac
ClusterVersion: Installing "4.14.0-0.okd-2023-11-14-101924" for 42 minutes: Unable to apply 4.14.0-0.okd-2023-11-14-101924: some cluster operators are not available
ClusterOperators:
        clusteroperator/authentication is not available (OAuthServerRouteEndpointAccessibleControllerAvailable: Get "https://oauth-openshift.apps.okd.ia5-f1.net/healthz": EOF) because OAuthServerRouteEndpointAccessibleControllerDegraded: Get "https://oauth-openshift.apps.okd.ia5-f1.net/healthz": EOF
        clusteroperator/console is not available (DeploymentAvailable: 0 replicas available for console deployment
RouteHealthAvailable: failed to GET route (https://console-openshift-console.apps.okd.ia5-f1.net/): Get "https://console-openshift-console.apps.okd.ia5-f1.net/": EOF) because RouteHealthDegraded: failed to GET route (https://console-openshift-console.apps.okd.ia5-f1.net/): Get "https://console-openshift-console.apps.okd.ia5-f1.net/": EOF
        clusteroperator/ingress is degraded because The "default" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: CanaryChecksSucceeding=False (CanaryChecksRepetitiveFailures: Canary route checks for the default ingress controller are failing)

- Oc get all -n openshift-console:

NAME                             READY   STATUS    RESTARTS      AGE
pod/console-685f8455f6-cptcf     0/1     Running   3 (35s ago)   10m
pod/console-767f768578-42l5c     0/1     Running   4 (16s ago)   13m
pod/downloads-557648459f-j8c9n   1/1     Running   0             16m

NAME                TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
service/console     ClusterIP   172.30.234.169   <none>        443/TCP   16m
service/downloads   ClusterIP   172.30.201.115   <none>        80/TCP    16m

NAME                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/console     0/1     1            0           16m
deployment.apps/downloads   1/1     1            1           16m

NAME                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/console-685f8455f6     1         1         0       10m
replicaset.apps/console-767f768578     1         1         0       13m
replicaset.apps/console-7b97749897     0         0         0       15m
replicaset.apps/console-8985cd7c8      0         0         0       16m
replicaset.apps/downloads-557648459f   1         1         1       16m

NAME                                 HOST/PORT                                         PATH   SERVICES    PORT    TERMINATION          WILDCARD
route.route.openshift.io/console     console-openshift-console.apps.okd.ia5-f1.net            console     https   reencrypt/Redirect   None
route.route.openshift.io/downloads   downloads-openshift-console.apps.okd.ia5-f1.net          downloads   http    edge/Redirect        None

- authentication:
NAME                                   READY   STATUS    RESTARTS   AGE
pod/oauth-openshift-5d479b8d64-7ft2v   1/1     Running   0          14m

NAME                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
service/oauth-openshift   ClusterIP   172.30.184.83   <none>        443/TCP   27m

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/oauth-openshift   1/1     1            1           17m

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/oauth-openshift-5d479b8d64   1         1         1       14m
replicaset.apps/oauth-openshift-b5c6d77f6    0         0         0       17m
replicaset.apps/oauth-openshift-c669664b8    0         0         0       17m

NAME                                       HOST/PORT                             PATH   SERVICES          PORT   TERMINATION            WILDCARD
route.route.openshift.io/oauth-openshift   oauth-openshift.apps.okd.ia5-f1.net          oauth-openshift   6443   passthrough/Redirect   None

- Oc describe authentication:
Name:                 oauth-openshift-5d479b8d64-7ft2v
Namespace:            openshift-authentication
Priority:             2000000000
Priority Class Name:  system-cluster-critical
Service Account:      oauth-openshift
Node:                 okd4-control-plane-1.okd.ia5-f1.net/192.168.22.6
Start Time:           Mon, 27 Nov 2023 20:56:51 +0100
Labels:               app=oauth-openshift
                      oauth-openshift-anti-affinity=true
                      pod-template-hash=5d479b8d64
Annotations:          k8s.v1.cni.cncf.io/network-status:
                        [{
                            "name": "openshift-sdn",
                            "interface": "eth0",
                            "ips": [
                                "10.128.0.88"
                            ],
                            "default": true,
                            "dns": {}
                        }]
                      openshift.io/scc: node-exporter
                      operator.openshift.io/bootstrap-user-exists: true
                      operator.openshift.io/rvs-hash: voCdTx8zzkxz5dXDL9o_rDpSIwsQrNqNcI1Cj4vVZlGFT5nPEHVSKoQ4-kayaOEuDL61tpkZqgSdeKxxowZVHw
Status:               Running
IP:                   10.128.0.88
IPs:
  IP:           10.128.0.88
Controlled By:  ReplicaSet/oauth-openshift-5d479b8d64
Containers:
  oauth-openshift:
    Container ID:  cri-o://70e870111a6cd5b0dd37b4665f7fc434131bd4a01f7e57b3a924d97c181df91a
    Image:         quay.io/openshift/okd-content@sha256:5a9959ba5918c57ff1a49dffa9893bf13fd032db8711118b062c558a5400545d
    Image ID:      quay.io/openshift/okd-content@sha256:5a9959ba5918c57ff1a49dffa9893bf13fd032db8711118b062c558a5400545d
    Port:          6443/TCP
    Host Port:     0/TCP
    Command:
      /bin/bash
      -ec
    Args:
      if [ -s /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt ]; then
          echo "Copying system trust bundle"
          cp -f /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
      fi
      exec oauth-server osinserver \
      --config=/var/config/system/configmaps/v4-0-config-system-cliconfig/v4-0-config-system-cliconfig \
      --v=2 \
      --audit-log-format=json \
      --audit-log-maxbackup=10 \
      --audit-log-maxsize=100 \
      --audit-log-path=/var/log/oauth-server/audit.log \
      --audit-policy-file=/var/run/configmaps/audit/audit.yaml

    State:          Running
      Started:      Mon, 27 Nov 2023 20:56:54 +0100
    Ready:          True
    Restart Count:  0
    Requests:
      cpu:        10m
      memory:     50Mi
    Liveness:     http-get https://:6443/healthz delay=30s timeout=1s period=10s #success=1 #failure=3
    Readiness:    http-get https://:6443/healthz delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:  <none>
    Mounts:
      /var/config/system/configmaps/v4-0-config-system-cliconfig from v4-0-config-system-cliconfig (ro)
      /var/config/system/configmaps/v4-0-config-system-service-ca from v4-0-config-system-service-ca (ro)
      /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle from v4-0-config-system-trusted-ca-bundle (ro)
      /var/config/system/secrets/v4-0-config-system-ocp-branding-template from v4-0-config-system-ocp-branding-template (ro)
      /var/config/system/secrets/v4-0-config-system-router-certs from v4-0-config-system-router-certs (ro)
      /var/config/system/secrets/v4-0-config-system-serving-cert from v4-0-config-system-serving-cert (ro)
      /var/config/system/secrets/v4-0-config-system-session from v4-0-config-system-session (ro)
      /var/config/user/template/secret/v4-0-config-user-template-error from v4-0-config-user-template-error (ro)
      /var/config/user/template/secret/v4-0-config-user-template-login from v4-0-config-user-template-login (ro)
      /var/config/user/template/secret/v4-0-config-user-template-provider-selection from v4-0-config-user-template-provider-selection (ro)
      /var/log/oauth-server from audit-dir (rw)
      /var/run/configmaps/audit from audit-policies (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-dsl2s (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  audit-policies:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      audit
    Optional:  false
  audit-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/log/oauth-server
    HostPathType:
  v4-0-config-system-session:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-system-session
    Optional:    false
  v4-0-config-system-cliconfig:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      v4-0-config-system-cliconfig
    Optional:  false
  v4-0-config-system-serving-cert:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-system-serving-cert
    Optional:    false
  v4-0-config-system-service-ca:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      v4-0-config-system-service-ca
    Optional:  false
  v4-0-config-system-router-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-system-router-certs
    Optional:    false
  v4-0-config-system-ocp-branding-template:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-system-ocp-branding-template
    Optional:    false
  v4-0-config-user-template-login:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-user-template-login
    Optional:    true
  v4-0-config-user-template-provider-selection:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-user-template-provider-selection
    Optional:    true
  v4-0-config-user-template-error:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  v4-0-config-user-template-error
    Optional:    true
  v4-0-config-system-trusted-ca-bundle:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      v4-0-config-system-trusted-ca-bundle
    Optional:  true
  kube-api-access-dsl2s:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
    ConfigMapName:           openshift-service-ca.crt
    ConfigMapOptional:       <nil>
QoS Class:                   Burstable
Node-Selectors:              node-role.kubernetes.io/master=
Tolerations:                 node-role.kubernetes.io/master:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists for 120s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 120s
Events:
  Type     Reason            Age                From               Message
  ----     ------            ----               ----               -------
  Warning  FailedScheduling  26m (x3 over 26m)  default-scheduler  0/1 nodes are available: 1 node(s) didn't match pod anti-affinity rules. preemption: 0/1 nodes are available: 1 node(s) didn't match pod anti-affinity rules..
  Normal   Scheduled         26m                default-scheduler  Successfully assigned openshift-authentication/oauth-openshift-5d479b8d64-7ft2v to okd4-control-plane-1.okd.ia5-f1.net
  Normal   AddedInterface    26m                multus             Add eth0 [10.128.0.88/23] from openshift-sdn
  Normal   Pulled            26m                kubelet            Container image "quay.io/openshift/okd-content@sha256:5a9959ba5918c57ff1a49dffa9893bf13fd032db8711118b062c558a5400545d" already present on machine
  Normal   Created           26m                kubelet            Created container oauth-openshift
  Normal   Started           26m                kubelet            Started container oauth-openshift
  Warning  ProbeError        26m                kubelet            Readiness probe error: Get "https://10.128.0.88:6443/healthz": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
body:
  Warning  Unhealthy  26m  kubelet  Readiness probe failed: Get "https://10.128.0.88:6443/healthz": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

- ingress:
NAME                                  READY   STATUS    RESTARTS      AGE
pod/router-default-7d8d9f5897-jflt2   1/1     Running   2 (21m ago)   27m

NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                   AGE
service/router-internal-default   ClusterIP   172.30.72.171   <none>        80/TCP,443/TCP,1936/TCP   27m

NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/router-default   1/1     1            1           27m

NAME                                        DESIRED   CURRENT   READY   AGE
replicaset.apps/router-default-7d8d9f5897   1         1         1       27m

- ingress canary:
NAME                       READY   STATUS    RESTARTS   AGE
pod/ingress-canary-4d7ck   1/1     Running   0          39m
pod/ingress-canary-527l2   1/1     Running   0          29m
pod/ingress-canary-55lls   1/1     Running   0          28m
pod/ingress-canary-pklbd   1/1     Running   0          28m

NAME                     TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
service/ingress-canary   ClusterIP   172.30.178.203   <none>        8080/TCP,8888/TCP   39m

NAME                            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
daemonset.apps/ingress-canary   4         4         4       4            4           kubernetes.io/os=linux   39m

NAME                              HOST/PORT                                             PATH   SERVICES         PORT   TERMINATION     WILDCARD
route.route.openshift.io/canary   canary-openshift-ingress-canary.apps.okd.ia5-f1.net          ingress-canary   8080   edge/Redirect   None

Does anyone have an idea what should be the problem ?
Thanks a lot for help.

**Comment by vrutkovs** [comment]
>I attached some logs:

Did must-gather produce the archive? That's the one we need

**Reply by MainMan1998** [reply]
Yes, it generated a folder, I attached it:
[must-gather.local.9159450260355983370.tar.gz](https://github.com/okd-project/okd/files/13523000/must-gather.local.9159450260355983370.tar.gz)

**Reply by vrutkovs** [reply]
slow etcd disk, a lot of messages like:
>2023-11-27T20:04:16.561658866Z {"level":"warn","ts":"2023-11-27T20:04:16.558225Z","caller":"etcdserver/util.go:170","msg":"apply request took too long","took":"368.104371ms","expected-duration":"200ms","prefix":"read-only range ","request":"key:\"/kubernetes.io/clusterroles/multus-admission-controller-webhook\" ","response":"range_response_count:1 size:740"}

**Reply by MainMan1998** [reply]
Thanks for your answer.
It's strange, the okd cluster run on ssd (8 ssd on raid 10).
I tried to re-install the cluster (with the same process), I increased the master ressources:
8 CPU / 40GB RAM / 128 GB SSD
but I have the same result, with sames errors.

I have one thing who I don't understand, on the redhat website, the authentication is down but when I saw pods status (with the oc commands), these ones are running =)

I attached the new installation logs:
[must-gather.local.4846047636554225253.tar.gz](https://github.com/okd-project/okd/files/13530399/must-gather.local.4846047636554225253.tar.gz)

**Reply by melledouwsma** [reply]
The console operator is unavailable because the ingress operator is degraded. The authentication operator is also degraded because of this. The ingress operator reports the canaries route checks are failing (`oc get co ingress -o yaml`):

```
  - lastTransitionTime: '2023-12-01T13:38:41Z'
    message: 'The "default" ingress controller reports Degraded=True: DegradedConditions:
      One or more other status conditions indicate a degraded state: CanaryChecksSucceeding=False
      (CanaryChecksRepetitiveFailures: Canary route checks for the default ingress
      controller are failing)'
    reason: IngressDegraded
    status: 'True'
    type: Degraded
```

I would first look into resolving that issue. For example, is the wildcard `*.apps.okd.ia5-f1.net` pointing to the correct DNS record? Are there any loadbalancers in between and are they functioning correctly, etc.

**Reply by MainMan1998** [reply]
I use a local dns based on named/bind, I don't have a wildcard, all certificates are self-signed
I have the record *.apps.okd.ia5-f1.net, this one is pointing on the oc bastion.
You will find the dns configuration:

```
opproomn01                  IN      A       192.168.22.4
; OpenShift Container Platform Cluster - A records
okd4-bootstrap.okd          IN      A       192.168.22.5
okd4-control-plane-1.okd    IN      A       192.168.22.6
;okd4-control-plane-2.okd    IN      A       192.168.22.7
;okd4-control-plane-3.okd    IN      A       192.168.22.8
okd4-control-compute-1.okd  IN      A       192.168.22.9
okd4-control-compute-2.okd  IN      A       192.168.22.10
okd4-control-compute-3.okd  IN      A       192.168.22.11

; OpenShift internal cluster IPs - A records
apps.okd                    IN      A       192.168.22.4
api.okd                     IN      A       192.168.22.4
api-int.okd                 IN      A       192.168.22.4
*.apps.okd                  IN      A       192.168.22.4
etcd-0.okd                  IN      A       192.168.22.6
console-openshift-console.apps.okd     IN     A     192.168.22.4
oauth-openshift.apps.okd    IN      A     192.168.22.4

; OpenShift internal cluster IPs - SRV records
_etcd-server-ssl._tcp.okd    86400     IN    SRV     0    10    2380    etcd-0.okd
```


The reverse zone:
```
4      IN      PTR     opproomn01.ia5-f1.net.

; OpenShift Container Platform Cluster - PTR records
5      IN      PTR     okd4-bootstrap.okd.ia5-f1.net.
6      IN      PTR     okd4-control-plane-1.okd.ia5-f1.net.
9      IN      PTR     okd4-compute-1.okd.ia5-f1.net.
10     IN      PTR     okd4-compute-2.okd.ia5-f1.net.
11     IN      PTR     okd4-compute-3.okd.ia5-f1.net.
```

I can ping and resolve the console:

```
ping console-openshift-console.apps.okd.ia5-f1.net
PING console-openshift-console.apps.okd.ia5-f1.net (192.168.22.4) 56(84) bytes of data.
64 bytes from opproomn01.ia5-f1.net (192.168.22.4): icmp_seq=1 ttl=64 time=0.118 ms
```

The first time who I installed the cluster, I installed it with 3 masters and I had access to the console and everything was working.
Actually, I would like to install it with one master, I set the dns to have one master + configuration installation file.

You will find bellow the file configuration:

```
apiVersion: v1
baseDomain: ia5-f1.net
metadata:
  name: okd

compute:
- hyperthreading: Enabled
  name: worker
  replicas: 0

controlPlane:
  hyperthreading: Enabled
  name: master
  replicas: 1

networking:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  networkType: OpenShiftSDN
  serviceNetwork:
  - 172.30.0.0/16

platform:
  none: {}

fips: false

pullSecret: 'pull secret'
sshKey: 'ssh key'
```

I attached the haproxy configuration:
```
# Global settings
#---------------------------------------------------------------------
global
    maxconn     20000
    log         /dev/log local0 info
    chroot      /var/lib/haproxy
    pidfile     /var/run/haproxy.pid
    user        haproxy
    group       haproxy
    daemon

    # turn on stats unix socket
    stats socket /var/lib/haproxy/stats

#---------------------------------------------------------------------
# common defaults that all the 'listen' and 'backend' sections will
# use if not designated in their block
#---------------------------------------------------------------------
defaults
    mode                    http
    log                     global
    option                  httplog
    option                  dontlognull
    option http-server-close
    option forwardfor       except 127.0.0.0/8
    option                  redispatch
    retries                 3
    timeout http-request    10s
    timeout queue           1m
    timeout connect         10s
    timeout client          300s
    timeout server          300s
    timeout http-keep-alive 10s
    timeout check           10s
    maxconn                 20000

listen stats
    bind :9000
    mode http
    stats enable
    stats uri /

frontend okd4_k8s_api_fe
    bind :6443
    default_backend okd4_k8s_api_be
    mode tcp
    option tcplog

backend okd4_k8s_api_be
    balance source
    mode tcp
    server      okd4-bootstrap 192.168.22.5:6443 check
    server      okd4-control-plane-1 192.168.22.6:6443 check

frontend okd4_machine_config_server_fe
    bind :22623
    default_backend okd4_machine_config_server_be
    mode tcp
    option tcplog

backend okd4_machine_config_server_be
    balance source
    mode tcp
    server      okd4-bootstrap 192.168.22.5:22623 check
    server      okd4-control-plane-1 192.168.22.6:22623 check

frontend okd4_http_ingress_traffic_fe
    bind :80
    default_backend okd4_http_ingress_traffic_be
    mode tcp
    option tcplog

backend okd4_http_ingress_traffic_be
    balance source
    mode tcp
    server      okd4-compute-1 192.168.22.9:80 check
    server      okd4-compute-2 192.168.22.10:80 check
    server      okd4-compute-3 192.168.22.11:80 check

frontend okd4_https_ingress_traffic_fe
    bind *:443
    default_backend okd4_https_ingress_traffic_be
    mode tcp
    option tcplog

backend okd4_https_ingress_traffic_be
    balance source
    mode tcp
    server      okd4-compute-1 192.168.22.9:443 check
    server      okd4-compute-2 192.168.22.10:443 check
    server      okd4-compute-3 192.168.22.11:443 check
```
I have no idea, normally if it works with 3 master, it should work with 1 master ?

**Reply by melledouwsma** [reply, marked as answer]
From the haproxy configuration:

```
backend okd4_https_ingress_traffic_be
    balance source
    mode tcp
    server      okd4-compute-1 192.168.22.9:443 check
    server      okd4-compute-2 192.168.22.10:443 check
    server      okd4-compute-3 192.168.22.11:443 check
```

From the must-gather:

```
$ omg get po -n openshift-ingress -o wide
NAME                             READY  STATUS   RESTARTS  AGE    IP            NODE
router-default-7d8d9f5897-bwrzv  1/1    Running  2         1h53m  192.168.22.6  okd4-control-plane-1.okd.ia5-f1.net
```

The cluster ingress is running on 192.168.22.6 and that node is not in the haproxy backend configuration. You can adjust the default  IngressController `replicas` and `nodePlacement` to schedule 3 pods on those nodes. You can find more on the ingress operator [here](https://docs.okd.io/4.14/networking/ingress-operator.html). Or you can adjust the haproxy setup, when only 1 replica would suffice for this cluster.

As a sidenote, the SRV records are no longer needed since OKD 4.4. Please see [User-provisioned DNS requirements](https://docs.okd.io/4.14/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-dns-user-infra_installing-platform-agnostic) for more on DNS settings in a UPI installation.

**Reply by MainMan1998** [reply]
It finally works, I add the master and delete the compute nodes on the haproxy, like this:
```
frontend okd4_http_ingress_traffic_fe
    bind :80
    default_backend okd4_http_ingress_traffic_be
    mode tcp
    option tcplog

backend okd4_http_ingress_traffic_be
    balance source
    mode tcp
    server      okd4-control-plane-1 192.168.22.6:80 check

frontend okd4_https_ingress_traffic_fe
    bind *:443
    default_backend okd4_https_ingress_traffic_be
    mode tcp
    option tcplog

backend okd4_https_ingress_traffic_be
    balance source
    mode tcp
    server      okd4-control-plane-1 192.168.22.6:443 check
```
Small question before closing the thread, do I need to put the compute nodes on the haproxy backend next to the control plane ?
I tried with 2 masters, just to know if it's working and I confirm, it's work.
Thanks a lot for help :)

**Reply by melledouwsma** [reply]
In this cluster — with only one control plane — the ingress operator will schedule the ingress pod on that control plane node by default. So you would not need to add the compute nodes to the haproxy backend. In a cluster with multiple control planes, the ingress controller default behaviour is different and will schedule the ingress pod(s) on the compute nodes.
